### PR TITLE
fix pan-os-list-pcaps and pan-os-get-pcaps bugs in panorama

### PR DESCRIPTION
**Pull request by GuyAfik**
## Status
- [ ] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: https://github.com/demisto/etc/issues/47887

## Description
This PR fixes several issues with the ```pan-os-get-pcaps``` and ```pan-os-list-pcaps```
1) a ```TypeError``` which is raised when there are no pcaps found in the firewall in the ```pan-os-list-pcaps``` command.
2) raise the api error when the pcap was not found in the the ```pan-os-get-pcaps```  instead of this error: ```PCAP download failed. Most likely cause is the file size limitation.\n'
            'For information on how to download manually, see the documentation for this integration.```
3) raise a correct error when sending the ```filter-pcap``` without the ```from``` argument. See [docs](https://docs.paloaltonetworks.com/pan-os/9-0/pan-os-panorama-api/pan-os-xml-api-request-types/export-files-api/export-packet-captures/export-threat-filter-and-data-filtering-pcaps)
4) improves the documentation of the ```pan-os-get-pcaps``` arguments to state which are required and to which filter type it is required.

## Screenshots
Paste here any images that will help the reviewer

## Must have
- [x] Tests
- [ ] Documentation

### CODE REVIEW

**Comment by GuyAfik** [comment]
Failed unit-tests - https://app.circleci.com/pipelines/github/demisto/content/156002/workflows/9ab1f94c-3492-444d-81a6-45d348d7e573/jobs/503737

**Comment by JudahSchwartz** [comment]
is this ready yet? We arent waiting on CS?

**Comment by GuyAfik** [comment]
> is this ready yet? We arent waiting on CS?

These are some issues I noticed when digging inside the code, its not related to the other bug.

File: Packs/PAN-OS/.secrets-ignore [hunk 1]
```diff
@@ -13,4 +13,5 @@ https://test_pb_task.not.real
 10.10.0.12
 10.10.0.0
 10.10.0.1
-10.10.0.1
\ No newline at end of file
+10.10.0.1
+capCo
```
Comment (abaumgarten) [inline Packs/PAN-OS/.secrets-ignore:17]: It is not a good practice to add this name here. Please change the name of your tests class instead.

Comment (GuyAfik) [reply Packs/PAN-OS/.secrets-ignore]: done, why does it recognize it as a secret anyway?

File: Packs/PAN-OS/Integrations/Panorama/Panorama_test.py [hunk 2]
```diff
@@ -719,6 +719,80 @@ def test_prettify_rule():
...
+           trying to download a pcap file.
+
+        Then -
+           make sure that the error message from the api is actually returned.
+        """
+        from Panorama import panorama_get_pcap_command
+        no_pcaps_response = MockedResponse(
+            text='<?xml version="1.0"?>\n<response status="error">\n  <msg>\n    '
+                 '<line>test.pcap not present</line>\n  </msg>\n</response>\n',
+            status_code=200,
```
Comment (JudahSchwartz) [inline Packs/PAN-OS/Integrations/Panorama/Panorama_test.py:765]: the api returns a 200 in this case?

File: Packs/PAN-OS/Integrations/Panorama/Panorama_test.py [hunk 3]
```diff
@@ -719,6 +719,80 @@ def test_prettify_rule():
...
+           trying to download a filter pcap file.
+
+        Then -
+           make sure that the error message which states that the 'from' argument should be returned is presented.
+        """
+        from Panorama import panorama_get_pcap_command
+        no_pcaps_response = MockedResponse(
+            text='<?xml version="1.0"?>\n<response status="error">\n  <msg>\n    '
+                 '<line>test.pcap not present</line>\n  </msg>\n</response>\n',
+            status_code=200,
```
Comment (JudahSchwartz) [inline Packs/PAN-OS/Integrations/Panorama/Panorama_test.py:788]: same question here

File: Packs/PAN-OS/Integrations/Panorama/Panorama_test.py (hunk 2 above)
Comment (GuyAfik) [reply Packs/PAN-OS/Integrations/Panorama/Panorama_test.py:765]: yes

File: Packs/PAN-OS/Integrations/Panorama/Panorama_test.py (hunk 3 above)
Comment (GuyAfik) [reply Packs/PAN-OS/Integrations/Panorama/Panorama_test.py:788]: > same question here

yes

File: Packs/PAN-OS/Integrations/Panorama/Panorama.py [hunk 4]
```diff
@@ -3369,6 +3372,10 @@ def panorama_get_pcap_command(args: dict):
 
     # due to pcap file size limitation in the product. For more details, please see the documentation.
     if result.headers['Content-Type'] != 'application/octet-stream':
+        json_result = json.loads(xml2json(result.text)).get('response', {})
+        if (json_result.get('@status') or '') == 'error':
+            errors = '\n'.join([error for error in (json_result.get('msg') or {}).values()])
```
[the remark this answers is not in the document]
Comment (GuyAfik) [reply Packs/PAN-OS/Integrations/Panorama/Panorama.py:3377]: The error response is in a form of dictionary, which means there are a possibility of more errors to occur. I didn't encounter yet a situation where there is more then one key/value in the dict. Your suggestion would work as well. ill change.

File: Packs/PAN-OS/ReleaseNotes/1_11_2.md [hunk 5]
```diff
@@ -0,0 +1,8 @@
+
+#### Integrations
+##### Palo Alto Networks PAN-OS
+- Fixed an issue where the **pan-os-list-pcaps** command failed when no pcaps are found in the firewall.
+- Fixed an issue where the **pan-os-get-pcaps** command failed on invalid error when pcap was not found.
+- Fixed an issue where **pan-os-get-pcaps** command failed on invalid error when executed with *filter-pcap* without the *from* argument.
```
Comment (JudahSchwartz) [inline Packs/PAN-OS/ReleaseNotes/1_11_2.md:6]:
```suggestion
- Fixed an issue where the **pan-os-get-pcaps** command failed on invalid error when executed with *filter-pcap* without the *from* argument.
```

File: Packs/PAN-OS/ReleaseNotes/1_11_2.md [hunk 6]
```diff
@@ -0,0 +1,8 @@
+
+#### Integrations
+##### Palo Alto Networks PAN-OS
+- Fixed an issue where the **pan-os-list-pcaps** command failed when no pcaps are found in the firewall.
+- Fixed an issue where the **pan-os-get-pcaps** command failed on invalid error when pcap was not found.
```
Comment (JudahSchwartz) [inline Packs/PAN-OS/ReleaseNotes/1_11_2.md:5]:
```suggestion
- Fixed an issue where the **pan-os-get-pcaps** command failed on an invalid error when pcap was not found.
```

**Review by JudahSchwartz — APPROVED** [review]
see last couple doc changes. Nice PR!

**Pull request merged**